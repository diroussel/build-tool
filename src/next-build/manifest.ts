import { readFileSync } from 'fs';

export type Manifest = Record<string, string>;

function readManifestSync(manifestPath: string): Manifest {
  return JSON.parse(readFileSync(manifestPath, 'utf8')) as Manifest;
}

export { readManifestSync };
