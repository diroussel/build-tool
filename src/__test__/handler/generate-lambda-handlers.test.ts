/* eslint-disable @typescript-eslint/no-base-to-string */
import File from 'vinyl';
import {
  generateLambdaHandler,
  generateHandlers,
} from '../../handler/generate-lambda-handler';
import { readManifestSync } from '../../next-build/manifest';

describe('generateLambdaHandler', () => {
  it('should match expected output', () => {
    const content = generateLambdaHandler(
      '/handlers/_error.js',
      '/pages/_error.js'
    );
    expect(content).toMatchSnapshot();
  });
});

describe('generateHandlers', () => {
  it('should match expected output', (done) => {
    const content = [];
    const manifest = readManifestSync(
      'src/__test__/testdata/test1-pages-manifest.json'
    );
    const stream = generateHandlers(manifest, false);
    stream.on('data', (file: File) => {
      content.push({
        handlerPath: file.relative,
        content: file.contents.toString(),
      });
    });
    stream.on('end', () => {
      expect(content).toMatchSnapshot();
      done();
    });
    stream.end();
  });

  it('should match expected output 2', (done) => {
    const content = [];
    const manifest = readManifestSync(
      'src/__test__/testdata/test2-pages-manifest.json'
    );
    const stream = generateHandlers(manifest, false);
    // eslint-disable-next-line sonarjs/no-identical-functions
    stream.on('data', (file: File) => {
      content.push({
        handlerPath: file.relative,
        content: file.contents.toString(),
      });
    });
    stream.on('end', () => {
      expect(content).toMatchSnapshot();
      done();
    });
    stream.end();
  });
});
