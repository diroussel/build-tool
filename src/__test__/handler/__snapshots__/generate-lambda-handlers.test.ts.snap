// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateHandlers should match expected output 1`] = `
Array [
  Object {
    "content": "const page = require(\\"./../../../pages/ras/journey1/check-your-answers.js\\");
const handlerFactory = require(\\"@sls-next/next-aws-lambda\\");
const handler = handlerFactory(page);

module.exports.render = handler;
",
    "handlerPath": "handlers/ras/journey1/check-your-answers.js",
  },
  Object {
    "content": "const page = require(\\"./../../pages/ras/journey1.js\\");
const handlerFactory = require(\\"@sls-next/next-aws-lambda\\");
const handler = handlerFactory(page);

module.exports.render = handler;
",
    "handlerPath": "handlers/ras/journey1.js",
  },
  Object {
    "content": "const page = require(\\"./../../../pages/ras/journey1/form-page.js\\");
const handlerFactory = require(\\"@sls-next/next-aws-lambda\\");
const handler = handlerFactory(page);

module.exports.render = handler;
",
    "handlerPath": "handlers/ras/journey1/form-page.js",
  },
  Object {
    "content": "const page = require(\\"./../../../pages/ras/journey1/result.js\\");
const handlerFactory = require(\\"@sls-next/next-aws-lambda\\");
const handler = handlerFactory(page);

module.exports.render = handler;
",
    "handlerPath": "handlers/ras/journey1/result.js",
  },
  Object {
    "content": "const page = require(\\"./../../../pages/ras/journey2/[pageId].js\\");
const handlerFactory = require(\\"@sls-next/next-aws-lambda\\");
const handler = handlerFactory(page);

module.exports.render = handler;
",
    "handlerPath": "handlers/ras/journey2/[pageId].js",
  },
  Object {
    "content": "const page = require(\\"./../pages/_error.js\\");
const handlerFactory = require(\\"@sls-next/next-aws-lambda\\");
const handler = handlerFactory(page);

module.exports.render = handler;
",
    "handlerPath": "handlers/_error.js",
  },
]
`;

exports[`generateLambdaHandler should match expected output 1`] = `
"const page = require(\\"./../pages/_error.js\\");
const handlerFactory = require(\\"@sls-next/next-aws-lambda\\");
const handler = handlerFactory(page);

module.exports.render = handler;
"
`;
